distrho/src/DistrhoPluginVST3.cpp

```diff
@@ -1065,6 +1065,9 @@ class PluginVst3
     {
         if (processing)
         {
+            delete[] fDummyAudioBuffer;
```

Comment:
I suggest to not even do this part at all.
Buffer is deleted on plugin destructor, should be fine then.

Reply:
That is correct, overlooked that.

Reply:
I was meaning the whole block actually.
dont recreate the buffer here, seems pointless

Reply:
Where should it be created? as detailed in the issue report, `setProcessing(false)` can be called before `process()` causing a crash. Maybe a better solution is reverting this? and:

1. Have `process()` check if `setProcessing(true)` was called and return early otherwise. We agree calling `setProcessing(false)` before `process()` does not make sense but the host is doing it.

2. Null check `fDummyAudioBuffer` before writing it during `process()` . Not sure of side effects.

Reply:
as I see it, but might be wrong, it should only be recreated during `setupProcessing`.
if the nominal buffer size changes at any point, the host is supposed to call that.

Reply:
Wait maybe I misunderstood the first comment here. Are you talking about leaving array init to `setupProcessing()`, strip `setProcessing()` of any memory mgmt, and leave deletion *only*  for the destructor (original version calls delete in two places).

In other words revert everything and remove the delete from `setProcessing(false)`.

Reply:
yes that is correct.
do you see any problems with that approach? it wastes a bit of memory, but it is comes as side effect from more VST3 weirdness

Reply:
Not at all, it is the simplest and also more defensive solution. Unfortunately the latter comes with a cost but not I'd say it's more a philosophical one. Will push shortly.

Reply:
FYI, this is ready and tested.